### PR TITLE
Added feature: update heat map on data change + fix linting

### PR DESCRIPTION
**Pull request by DonovanDeSmedt**
When prop updateHeatmap is set to true --> map will rerender.
- New render of <GoogleMapReact> and data is changed, prop updateHeatmap should be true
- New render of <GoogleMapReact> and data is not changed, prop updateHeatmap should be false

### CODE REVIEW

**Comment by itsmichaeldiego** [comment]
@DonovanDeSmedt Great work dude! Do you mind adding a quick video showing it works?

@ZAKdev as you were the one who added heatmap and you work often with it, do you mind taking a quick look?

**Comment by DonovanDeSmedt** [comment]
@itsmichaeldiego thx, was quiet busy last week, I'll make a video tomorrow.

File: src/google_map.js [hunk 1]
```diff
@@ -299,9 +299,11 @@ export default class GoogleMap extends Component {
     }
 
     if (
-      !this._isCenterDefined(this.props.center) &&
-      this._isCenterDefined(nextProps.center)
+      (!this._isCenterDefined(this.props.center) &&
+        this._isCenterDefined(nextProps.center)) ||
+      this.props.updateHeatmap
```
[the remark this answers is not in the document]
Comment (itsmichaeldiego) [reply src/google_map.js]: Also, I am a bit worried, my dear friend @istarkov left a comment [here](https://github.com/google-map-react/google-map-react/pull/593/files#diff-0146a78a4269b3d2562b8f44d0137200R479) that map should be initialized only once, I am pretty sure there is a way to achieve updating the map without re-initializing it, this also will affect the performance of your app, would you like to work on this together to find the best way to do it?

**Comment by itsmichaeldiego** [comment]
@DonovanDeSmedt BTW, I just fixed the build and in order to do so I've to fix the linter, so you might have to rebase and remove the fix linting comment. Let me know if you don't have time and I can do it for you.

**Comment by DonovanDeSmedt** [comment]
Feel free to do it for me, I messed up some things locally :(

File: src/google_map.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (DonovanDeSmedt) [reply src/google_map.js]: Yeah, that's what I thought. It's a quick 'fix' for the moment but there should be a better way to implement this. I'll try to look into it.

**Comment by itsmichaeldiego** [comment]
@DonovanDeSmedt Great work man! I will create a minor release soon

**Comment by itsmichaeldiego** [comment]
@DonovanDeSmedt Great work man! I might need you to add it to the documentation so users know that this is a possibility! (To use this libraries). And I will create a minor release asap.

**Comment by DonovanDeSmedt** [comment]
@itsmichaeldiego Indeed, that's necessary. I'll update the documentation in the readme.

**Comment by itsmichaeldiego** [comment]
@DonovanDeSmedt Thanks! Looking forward for it!

**Comment by DonovanDeSmedt** [comment]
@itsmichaeldiego  Can you check the updated readme to verify there is enough information about the change in this pr?

**Comment by itsmichaeldiego** [comment]
@DonovanDeSmedt Looks great, thanks man! I will merge it soon

**Comment by itsmichaeldiego** [comment]
@DonovanDeSmedt Merged and deployed in version `1.1.0`, please try it and let me know

**Comment by itsmichaeldiego** [comment]
@DonovanDeSmedt  This created an issue #673 

Let me know if you can take a look at that

**Comment by DonovanDeSmedt** [comment]
> @DonovanDeSmedt This created an issue: #673
> 
> Let me know if you can take a look at that, because the older way of using libraries stopped working
> 
> Maybe we should change the approach to use heatmapLibrary the same way we used to add the libraries in the URL:
> 
> ```
> <GoogleMapReact
>   bootstrapURLKeys={{
>     key: 'xxxxxxxxxxxxxx',
>     libraries: ['places', 'drawing', 'heatmap']
>   }}
>   yesIWantToUseGoogleMapApiInternals
>   onGoogleApiLoaded={({ maps }) => {
>     console.log(maps.places);
>     console.log(maps.drawing);
>   }}
> />
> ```

Ok, I'll update it, by the weekend at the latest.

**Comment by snhasani** [comment]
```
drawingLibrary={true}
placesLibrary ={true}
```
OMG! I can't believe you changed an important API without any notice, updating docs and release log. This is super critical and ppl are using this component in a serious business and product. Please bring back the wrong way, anyway.

**Comment by DonovanDeSmedt** [comment]
The new implemenation of loading libraries was added to be consistent with the heatmap library (as described in the [api docs](https://github.com/google-map-react/google-map-react/blob/master/API.md#example)). 
```
drawingLibrary={true} 
```


This implementation no longer supported the previous way  
``` 
bootstrapURLKeys={{
    key: 'xxxxxxxxxxxxxx',
    libraries: ['places', 'drawing', 'heatmap']
  }} 
``` 

It was released in release [1.1.0](https://github.com/google-map-react/google-map-react/releases/tag/1.1.0) but it should indeed be backward compatible. Hence my [pr](https://github.com/google-map-react/google-map-react/pull/681) to fix this issue (and support both ways) if that's wanted.

**Comment by itsmichaeldiego** [comment]
Thanks @DonovanDeSmedt! Great input, I already answered in your PR, I think the heatmap change was created a few months back by @ZAKdev and I did not remember that we already had another way to handle libraries, that was my mistake. I would keep it the old way, just importing libraries with the array.

**Comment by itsmichaeldiego** [comment]
@snhasani We fixed this and released the fix in `1.1.1`

**Pull request merged**